### DISCUSSION TITLE
cause of "/_next/webpack-hmr failed"

### DISCUSSION THREAD

**Discussion by birthdaycorp** [vercel/next.js#34986] [Help]
My app's WebSockets connection fails in development mode on port 3000. If I use the app on port 64729, it works. 
To figure out why, I run `vercel dev --debug` in terminal and get this when I access the app on port 3000:

```
> [debug] [2022-03-03T02:19:24.690Z] GET /_next/webpack-hmr
> [debug] [2022-03-03T02:19:24.690Z] Reading `package.json` file
> [debug] [2022-03-03T02:19:24.690Z] Reading `vercel.json` file
.....
.....
> [debug] [2022-03-03T02:19:24.704Z] Using local env: /Users/me/code/app/.env
> [debug] [2022-03-03T02:19:24.707Z] Requiring runtime: "/Users/me/Library/Caches/com.vercel.cli/dev/builders/node_modules/@vercel/static-build"
> [debug] [2022-03-03T02:19:24.709Z] Checking build result's 1 `routes` to match /_next/webpack-hmr
> [debug] [2022-03-03T02:19:24.709Z] Found matching route http://localhost:64729/_next/webpack-hmr for /_next/webpack-hmr
> [debug] [2022-03-03T02:19:24.709Z] ProxyPass: http://localhost:64729/_next/webpack-hmr
> [debug] [2022-03-03T02:19:31.237Z] Detected "upgrade" event, but closing socket because no frontend dev server is running
```

So from the looks of it, WebSockets works, it's just not working on port 3000. How can I make it work on that port? Is this because of my .env file? Thank you!!!! :)

**Comment by quantafire** [comment]
I have the same issue. This ticket helped me find the port number it was running on and now it works if I access it directly.

In other [tickets](https://github.com/vercel/next.js/issues/30491), proxying the dev server through Nginx caused issues but it seems that the latest version of `vercel` does some sort of proxy that no longer works for my repo either when I upgrade to v12+
